### ISSUE TITLE
File path which contains .(dot) causes error of dvipdfmx

### ISSUE THREAD

**Issue by kn1cht** [texjporg/ptex2pdf#12]
# TL;DR
Command like `ptex2pdf -l foo.bar/test` would fail because characters after `'.'` removed from the file path.

# How to Reproduce
Put .tex file in a directory that contains `'.'`. `dvipdfmx` tries to process `foo.dvi` and stops with error.

```bash
$ tree
.
└── foo.bar
    └── test.tex

$ ptex2pdf -l foo.bar/test
This is ptex2pdf[.lua] version 20180514.0.
Processing foo.bar/test
# ...
Output written on test.dvi (1 page, 300 bytes).
Transcript written on test.log.
foo.dvi -> foo.pdf

dvipdfmx:fatal: Could not open specified DVI (or XDV) file: foo.dvi

Output file removed.
ptex2pdf processing of foo.bar/test failed.
```

# Detail
I built my project with Visual Studio Code and [LaTeX Workshop](https://marketplace.visualstudio.com/items?itemName=James-Yu.latex-workshop) extension, and got above error.
Strangely, `ptex2pdf` tried to convert `github.dvi` to `github.pdf`.

LaTeX Workshop seemed to pass full pathname to `ptex2pdf`(excluding file extension).
I use [`ghq`](https://github.com/motemen/ghq) to manage my repositories, so the full path was like `$HOME/.ghq/github.com/user/repo/file.tex`.

I checked source of `ptex2pdf` and noticed that code below removes characters after the last dot of the file path.

https://github.com/texjporg/ptex2pdf/blob/e744cef162b4f8dc5b652dc6271bb9ffddf33dd8/ptex2pdf.lua#L400-L403

`.com/user/repo/file` in `$HOME/.ghq/github.com/user/repo/file` was regarded as an extension, so `github.dvi` was passed to `dvipdfmx`.

# Workaround
I could avoid this problem by ignoring dot before slash.

```diff
@@ -399,7 +399,7 @@ else
     end
   else
     -- if it has already an extension, we need to drop it to get the dvi name
-    bname = string.gsub(filename, "^(.*)%.[^.]+$", "%1")
+    bname = string.gsub(filename, "^(.*)%.[^./]+$", "%1")
   end
   -- filename may contain "/", but the intermediate output is written
   -- in current directory, so we need to drop it
```

Because I am not familiar with LaTeX nor lua, please let me know if you noticed anything about this solution.
